src/coreclr/vm/jitinterface.cpp

```diff
@@ -8804,7 +8804,15 @@ void CEEInfo::getFunctionEntryPoint(CORINFO_METHOD_HANDLE  ftnHnd,
     // Resolve methodImpl.
     ftn = ftn->GetMethodTable()->MapMethodDeclToMethodImpl(ftn);
 
-    ret = (void *)ftn->TryGetMultiCallableAddrOfCode(accessFlags);
+    if (!ftn->IsFCall() && ftn->MayHavePrecode() && ftn->GetPrecodeType() == PRECODE_FIXUP)
```

Comment:
I think this condition is not right for the tiered compilation disabled case (not default, but still used by some first parties). Does this need to call into code version manager?

Reply:
It is possible (although all the tests pass with tiered compilation disabled except for one where the failure looks unrelated to this). What do you think is the specific problem here?

Reply:
I think that the tiered compilation disabled case will go through unnecessary indirection when the target method is JITed already.

Reply:
I've tried to add `!ftn->HasStableEntryPoint()` to the condition. It severely degraded the plaintext benchmark performance. I've also thought that `!ftn->IsPointingToStableNativeCode()` might be needed, but the function seems to never report true together with the preexisting condition (based on my testing when I've added an assert and none of the coreclr pri1 tests hit it, with or without tiered compilation enabled).

Reply:
I think the logic should be something like this:
```
if (ftn->IsVersionable())
{
    IAT_PVALUE // must go via indirection to enable versioning 
}
else
{
    if (the target has final code)
        IAT_VALUE for the final code // this includes FCalls and method that are already JITed (w/o tiering)
    else
        IAT_PVALUE // must go via indirection to pick up the final code once it is ready
}
```

Reply:
I will give it a try